### PR TITLE
Feat handle overlord disconnection

### PR DESCRIPTION
**Pull request by calniotrey**
This is based on the other PR.

Changed the logic of the monthly pulsed event for corps:
Now there is a country event that fires itself every 30 days for each corp. It is fired at first by the creation of the corp.
Like the previous one, it fires a country event to handle the whole "build ships/deposits/..." logic between 5 and 25 days later and one country event to compute the taxes (this one won't happen every 1th month anymore, but every 30 days)

In those 3 events, the hq is already set (from the creation effect) => no need to get it
If the hq's owner is not the overlord (and if it accepts corps), set it as a new overlord
If there is no owner or if the owner doesn't accept corps, destroy the corp

This handles therefore the cases where:
- the planet is transfered to a new owner => is set as subject
- the planet is transfered to a new owner that is already a vassal => corp gets destroyed
- the planet has no owner anymore => corp gets destroyed

### CODE REVIEW

File: events/iisc_corps_events.txt [hunk 1]
```diff
@@ -408,3 +408,36 @@ country_event = {
...
+				limit = {
+					is_country_type = iisc_space_corporation
+					# which hq isn't owned by the new overlord
+					NOT = {
+						prev = {
+							any_owned_planet = {
+								has_planet_flag = iisc_hq_of_@prevprev
+							}
+						}
+					}
```
Comment (fireinthedisco) [inline events/iisc_corps_events.txt:434]:
```
NOT = {
    iisc_is_corp_connected = yes
}
```
?

File: common/scripted_effects/iisc_corps_effects.txt [hunk 2]
```diff
@@ -47,6 +42,178 @@ iisc_get_overlord = {
...
+							has_planet_flag = iisc_hq_of_@prevprevprev
+						}
+						save_event_target_as = iisc_hq_planet
+						remove_planet_flag = iisc_hq_of_@prevprevprev
+						prevprevprev = {
+							iisc_remove_hq_deposit_from_corp = yes
+						}
+					}
+				}
+			}
```
Comment (fireinthedisco) [inline common/scripted_effects/iisc_corps_effects.txt:125]: Some of these checks seem ultimately unnecessary, that is, scoping to the overlord, then a random subject (the former owner of the corporation), then a random planet of theirs, to get to the HQ? This actually seems needlessly restrictive, since these checks mean we can _only_ use this effect for removing a corporation when their owner has becoming a vassal and the corp is now invalid. But what if in the future we decide we would like to remove a corp and have their money distributed in some other scenario (where their owner hasn't been vassalized)?

In short it seems to me that this effect could just be done with a `random_planet` with the `iisc_hq_of_@` flag. This is slower but this event isn't running regularly, so I'm not sure it would be a serious issue.

File: events/iisc_pulse_events.txt [hunk 3]
```diff
@@ -128,23 +143,32 @@ event = {
...
 	id = iisc_pulse.21
 	is_triggered_only = yes
 	hide_window = yes
 	trigger = {
-		exists = overlord
+		exists = event_target:iisc_hq_planet
+		NOR = {
+			has_global_flag = iisc_stopped
+			has_global_flag = iisc_paused
+		}
```
Comment (fireinthedisco) [inline events/iisc_pulse_events.txt:168]: Where is `iisc_stopped` added? How is it distinct from `iisc_paused`?

File: events/iisc_pulse_events.txt [hunk 4]
```diff
@@ -101,23 +106,33 @@ country_event = {
 	}
 }
 
-# monthly pulse
+# monthly pulse but fires itself
```
Comment (fireinthedisco) [inline events/iisc_pulse_events.txt:109]: not monthly pulse at all any more

File: events/iisc_pulse_events.txt [hunk 5]
```diff
@@ -101,23 +106,33 @@ country_event = {
...
-					has_global_flag = iisc_paused
+				OR = {
+					NOT = {
+						exists = event_target:iisc_hq_planet
+					}
+					has_global_flag = iisc_stopped
 				}
 			}
-			every_country = {
+			# TODO destroy the country and cleanup
```
Comment (fireinthedisco) [inline events/iisc_pulse_events.txt:126]: Do we need todo here? We have the effect to destroy the country

**Review by fireinthedisco — APPROVED** [review]
If my understanding of all of this is correct, this could lead to some interesting situations - for example, a HQ planet gets conquered, the corp changes owners, and it owns a large number of stations still within the territory of its former owner.

I do not personally have an issue with a corp having stationary assets in territory that does not belong to it/its owner, in fact I think this should be something we specifically allow in the future (perhaps through particular interactions, like a corp asking a foreign government for rights to build in its territory?) but I believe that at the current moment, this changing of corporate ownership is the only way it could come about.

In the future, I think we should possibly make much of what happens in this PR be at the discretion of the player, based on interactions with the corporate sector. That is, you acquire a new planet and must decide whether you want to leave the corporation there intact, or nationalize it as spoils of war. Again though, that's just a future thing. What do you think?

For now, this looks good.

File: events/iisc_corps_events.txt (hunk 1 above)
Comment (calniotrey) [reply events/iisc_corps_events.txt:434]: :+1:

File: events/iisc_pulse_events.txt (hunk 3 above)
Comment (calniotrey) [reply events/iisc_pulse_events.txt:168]: Nowhere for now. This will need more PR ^^
The logic I thought of:
paused => momentarily paused (no corps spawn, less computations, ...)
stopped => prepare for desintall (kill the corps, their ships, ... stop the recurring events)

File: events/iisc_pulse_events.txt (hunk 5 above)
Comment (calniotrey) [reply events/iisc_pulse_events.txt:126]: Looks like I forgot it haha
